### ISSUE TITLE
Segmentation result display

### ISSUE THREAD

**Issue by MaxEAB** [Deci-AI/super-gradients#985]
Hi, Is there a sample code available for overlaying the predicted mask on the image for a visual verification? I am using the pretrained PP_LITE_T_SEG50 model and testing it out on one of the "cityscape" images.

**Comment by BloodAxe** [comment]
At this point we haven't made predict() functionality for segmentation models. However it's on the roadmap but work has not been started yet. If you would like to contribute - let us know and we can assist with making a PR for this feature.

**Comment by MaxEAB** [comment]
Thank you, will take a look at those.

Do you have code available that would help train the PP_LITE_T_SEG50 model on my own custom data?

**Comment by Louis-Dupont** [comment]
You can check out this tutorial: [transfer learning - semantic segmentation](https://github.com/Deci-AI/super-gradients/blob/master/tutorials/SG_transfer_learning_semantic_segmentation.ipynb)
If you still struggle, don't forget to check out the documentation website, it should also include some extra information :)

**Comment by MaxEAB** [comment]
Great, thank you!

Is this the documentation? -- https://docs.deci.ai/super-gradients/documentation/source/welcome.html

Regrading the overlay: Since the trained model has 19 classes, the mask is of size [1, 19, w, h]. Should I pick up the sigmoid argmax at each pixel location and also threshold at 0.5?

**Comment by BloodAxe** [comment]
For semantic segmentation you want to apply argmax to outputs (It is irrelevant whether you apply softmax or not) to get hard labels.

**Comment by MaxEAB** [comment]
Won't I still need a threshold (and thus a sigmoid) for this? Unless "unlabeled background" is one of the 19 classes in the training?

**Comment by BloodAxe** [comment]
I believe "unlabeled background" is not included as part of model's predictions output.
"unlabeled background" usually used to indicate areas that are not annotated and thus not contributing to loss during training. However there could still be valid objects in it, maybe too dark or too complex to annotate. 
During evaluation, predictions of the model in these regions are ignored and do not contribute to metric.

**Comment by MaxEAB** [comment]
> During evaluation, predictions of the model in these regions are ignored and do not contribute to metric.

What would the predicted mask be for these regions during inference? (I assumed it will get one of the 19 labels, and I that I need to use the sigmoid and threshold to weed those out.)

**Comment by jonasdieker** [comment]
The original cityscapes dataset has 32 classes. Is there some documentation which clarifies what these 19 classes correspond to?

**Comment by BloodAxe** [comment]
We have upcoming predict() method for segmentation models https://github.com/Deci-AI/super-gradients/pull/1320
And merged PR for visualizing segmentation predictions during training - https://github.com/Deci-AI/super-gradients/pull/1282
This should be available in 3.2
